### PR TITLE
fix: not enough capacity (Container::queryMaxCount() bug)

### PR DESCRIPTION
**Pull request by gpedro**
based on otland/forgottenserver#2953 by @nekiro

### CODE REVIEW

**Review by jlcvp — APPROVED** [review]
LGTM

File: src/player.cpp [hunk 1]
```diff
@@ -2565,23 +2565,30 @@ ReturnValue Player::queryAdd(int32_t index, const Thing& thing, uint32_t count,
...
-		}
+	//check if enough capacity
+	if (!hasCapacity(item, item->getItemCount())) {
+		return RETURNVALUE_NOTENOUGHCAPACITY;
+	}
 
-		if (!g_moveEvents->onPlayerEquip(const_cast<Player*>(this), const_cast<Item*>(item), static_cast<slots_t>(index), true)) {
-			return RETURNVALUE_CANNOTBEDRESSED;
-		}
+	ret = g_moveEvents->onPlayerEquip(const_cast<Player*>(this), const_cast<Item*>(item), static_cast<slots_t>(index), true);
```
Comment (oalbsilva) [inline src/player.cpp:2577]: Was a error in this line:
ret is enum ReturnValue, and onPlayerEquip returns a int.

i Change to: 
`ret = ReturnValue(g_moveEvents->onPlayerEquip(const_cast<Player*>(this), const_cast<Item*>(item), static_cast<slots_t>(index), true));`

**Review by oalbsilva — CHANGES_REQUESTED** [review]

**Comment by oalbsilva** [comment]
It is working correctly!

File: src/player.cpp (hunk 1 above)
Comment (oalbsilva) [reply src/player.cpp:2577]: The suggested change caused a problem, where players are no longer able to equip items.
So... I back to original code: 

```cpp
if (!g_moveEvents->onPlayerEquip(const_cast<Player*>(this), const_cast<Item*>(item), static_cast<slots_t>(index), true)) {
		return RETURNVALUE_CANNOTBEDRESSED;
}
```

Now its working!

**Review by FakeShinoda — APPROVED** [review]
LGTM

**Comment by nekiro** [comment]
Hey, there was already another fix for that pr, because it introduced another bug.
If you want to merge this change then please take this pr: https://github.com/otland/forgottenserver/pull/3064

**Comment by oalbsilva** [comment]
> Hey, there was already another fix for that pr, because it introduced another bug.
> If you want to merge this change then please take this pr: [otland/forgottenserver#3064](https://github.com/otland/forgottenserver/pull/3064)


It is a very difficult scenario to simulate...
But I ran into this bug last night!

**Comment by dudantas** [comment]
> Hey, there was already another fix for that pr, because it introduced another bug.
> If you want to merge this change then please take this pr: [otland/forgottenserver#3064](https://github.com/otland/forgottenserver/pull/3064)

Thanks for the feedback

**Review by dudantas — CHANGES_REQUESTED** [review]
According to the PR creator himself, this created another bug, which can be corrected by pulling the new PR from the link below:

https://github.com/otland/forgottenserver/pull/3064

**Pull request closed without merging**